plugins/announcements-backend/src/plugin.ts

```diff
@@ -0,0 +1,30 @@
+import {
+  createBackendPlugin,
+  coreServices,
+} from '@backstage/backend-plugin-api';
+import { createRouter } from './service/router';
+
+export const announcementsPlugin = createBackendPlugin({
+  pluginId: 'announcements',
+  register(env) {
+    env.registerInit({
+      deps: {
+        logger: coreServices.logger,
+        http: coreServices.httpRouter,
+        permissions: coreServices.permissions,
+      },
+      async init({ http, permissions }) {
+        http.use(
+          await createRouter({
+            persistenceContext: {
+              // TODO: How to reuse/refactor announcementsContext
+              announcementsStore: undefined,
```

Comment:
@kurtaking I just started to follow this [migration guide](https://backstage.io/docs/backend-system/building-plugins-and-modules/migrating) and look after few [examples](https://github.com/backstage/backstage/issues/18301). I didn't find something that has the same behavior because of the `persistenceContext` used in the plugin.

Do you have an idea what we can do to migrate this part ?

Reply:
Hey, I am looking into this now and think we need to do something like

```ts
persistenceContext: await initializePersistenceContext(database)
```